### PR TITLE
Issue #405 - Adds "How it Works" to README.rst

### PR DESCRIPTION
**Pull request by richmahn**
What I have done so far.  Still needs some revising and more specifics.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -32,10 +32,108 @@ Issue for its creation at https://github.com/unfoldingWord-dev/door43.org/issues
...
+Request Conversion Job
+----------------------
+
+Using the Pipeline and the corresponding numbers above, this describes each part of the pipeline and how each are integrated, both with each other as well as the AWS Services that are used.
+
+NOTE: This gives URLs and bucket names for development. For production, remove the `test-` prefix from domain or bucket name
+
+1. Gogs (Git website)
+
+When a repository is updated on `Gogs <https://test.door43.org:3000>`_, the commit triggers all webhooks in the repo's settings to be triggered. One of those webhooks, which our copy of Gogs sets up automatically for every new repo, is a call to https://test-api.door43.org/client/webhook (API Gateway -> Lambda function).
```
Comment (phillip-hopper) [inline README.rst]: "triggers...to be triggered" make no sense. Would be better if "to be triggered" was removed.

File: README.rst [hunk 2]
```diff
@@ -32,10 +32,108 @@ Issue for its creation at https://github.com/unfoldingWord-dev/door43.org/issues
...
+* gogs_url* - the URL to the Gogs site to verify user token (e.g. https://test.door43.org:3000)
+* gogs_user_token* - a user token of a valid user to prove they are a user so we can track job requests (for the webclient we just have one user token for all requests, given by the API Gateway)
+
+*these variables are set up in the `'client' Stage Variables <https://us-west-2.console.aws.amazon.com/apigateway/home?region=us-west-2#/apis/94c6v76xoh/stages/client>`_, so dev and prod gateways can have different variables
+
+The client_webhook function is responsible for standardizing both a manifest.json file and the resource containers from all types of repos committed to Gogs. Converters (#4) expect the files to be converted to be in a flat-level zip file, where all files to be converted (with the input file extension) are one file per chapter (Bible, OBS) and in alphabetical order for logging and display purposes. Once the files are zipped up and the zip file put at https://test-cdn.door43.org/temp/<repo>/<commit>, the client webhook function requests a job by posting a request to https://test-api.door43.org/tx/job and exits.
```
Comment (phillip-hopper) [inline README.rst:80]: I think `manifest.json` is changing to `manifest.yaml` soon.

File: README.rst [hunk 3]
```diff
@@ -32,10 +32,108 @@ Issue for its creation at https://github.com/unfoldingWord-dev/door43.org/issues
...
+
+3. Request Job (Lamdbda function - API Gateway triggered)
+
+Request Job is triggered through a call to the AWS API Gateway which triggers the `request_job lambda function <https://github.com/unfoldingWord-dev/tx-manager-lambda/blob/develop/functions/tx-manager_request_job/main.py>`_. This function expects the following variables in the payload:
```
Comment (phillip-hopper) [inline README.rst]: "triggered...which triggers", should be reworded, too many triggers.

File: README.rst [hunk 4]
```diff
@@ -32,10 +32,108 @@ Issue for its creation at https://github.com/unfoldingWord-dev/door43.org/issues
...
+
+* gogs_url* - the URL to the Gogs site to verify user token (e.g. https://git.door43.org)
+* api_url* - the base URL to the tX Manager API (e.g. https://test-api.door43.org)
+* data - information about the job to performed. It contains the following variables:
+
+   * gogs_user_token - a user token of a valid user to prove they are a user so we can traob requests
```
Comment (phillip-hopper) [inline README.rst]: sp: "traob"?

File: README.rst [hunk 5]
```diff
@@ -32,10 +32,108 @@ Issue for its creation at https://github.com/unfoldingWord-dev/door43.org/issues
...
+7. Deploy to Door43  (Lamdbda function - S3 modified file triggered)
+
+The uploading of build_log.json in #6 triggers the `Door43 Deploy function <<https://github.com/unfoldingWord-dev/tx-manager-lambda/blob/develop/functions/door43_deploy/main.py>`_.
+
+The Door43 Deploy function is what moves the HTML files converted by #5 and placed in the CDN bucket in #6 to live.door43.org and templates it based on the `door43.org layouts <https://github.com/unfoldingWord-dev/door43.org/tree/develop/_layouts>`_. It also generates header, status and navigation portions of the pages for each revision.
+
+Register Conversion Module
+--------------------------
+
+In order for tX Manager to know about a conversion module and to assign a conversion request to the module, it must be registered. To register a module, it must made a call to the API Gateway with the URL https://test-api.door43.org/tx/register. It expects
```
Comment (phillip-hopper) [inline README.rst]: typo: "it must made a call"

**Review by phillip-hopper — CHANGES_REQUESTED** [review]

File: README.rst (hunk 2 above)
Comment (richmahn) [reply README.rst:80]: Yeah, but in the repos the web client gets. I wonder if it should still compile a manifest.json file from the different manifest types. If we want to read the manifest on life.door43.org, should it be in JSON format or YAML? Anyway, for now it compiles a manifest.json file.

File: README.rst (hunk 4 above)
Comment (richmahn) [inline README.rst]: I think I lost a block of text. The next line should be api_url, which vanished. Will add.

**Review by phillip-hopper — APPROVED** [review]

**Pull request merged**